index.js

```diff
@@ -37,9 +45,73 @@ app.post('/users', async (req, res) => {
   console.log('ADDING NEW USER: ', name);
   const { data, error } = await db.insertUser(name);
   console.log('RETURNED DATA: ', data, error);
-  if(error) {
+  if (error) {
     res.status(400).json({ error });
   } else {
     res.status(200).json({ data });
   }
 });
+
+app.get('/auth/login', async (req, res) => {
+  const config = {
+    client_id: '0oa2w7lue0U6fnn3N5d7',
+    redirect_uri: 'http://localhost:3000/callback',
+    authorization_endpoint: 'https://dev-88956181.okta.com/oauth2/default/v1/authorize',
+    request_scopes: 'openid',
+  };
+
+  // Create and store a random "state" value
+  state = generateRandomString();
+  // Create and store a new PKCE code_verifier (the plaintext random secret)
+  codeVerifier = generateRandomString();
+
+  const codeChallenge = await pkceChallengeFromVerifier(codeVerifier);
+
+  const myUrl = `${config.authorization_endpoint}?response_type=${encodeURIComponent('code')}&client_id=${encodeURIComponent(config.client_id)}&state=${encodeURIComponent(state)}&scope=${encodeURIComponent(config.request_scopes)}&redirect_uri=${encodeURIComponent(config.redirect_uri)}&code_challenge=${encodeURIComponent(codeChallenge)}&code_challenge_method=${encodeURIComponent('S256')}`;
+
+  res.send({ url: myUrl });
+  res.end();
+});
+
+// After user signs into Authorization Server, the server will make a GET request
+// To our callback (localhost:5000/token)
+// This will hit this endpoint, where we want to 
+// Make a POST request to /token,
+app.get('/token', async (req, res) => {
+  const tokenEndpoint = 'https://dev-88956181.okta.com/oauth2/default/v1/token';
+  const clientId = '0oa2w7lue0U6fnn3N5d7';
```

Comment:
since this is used in multiple places, may be worth creating a variable for it at the top of the file